opencolorio_config_aces/clf/transforms/arri/generate.py

```diff
@@ -0,0 +1,445 @@
+        Whether to print additional informative text.
+
+    Returns
+    -------
+    ocio.LogCameraTransform
+         *OpenColorIO* `LogCameraTransform`.
+    """
+
+    if not (160 <= ei <= 1280):
+        print(
+            f"Error: Unsupported EI{ei:d} requested, must be 160 <= EI <= 1280".format(
+                ei
+            )
+        )
+        return False
+
+    # v3_IDT_maker.py parameters
+    nominalEI = 400
+    midGraySignal = 0.01
+    encodingGain = 500 / 1023 * 0.525  # was: 0.256598
+    encodingOffset = 400 / 1023  # was: 0.391007
+    cutPoint = 1 / 9  # renamed from "cut"
+    slope = 1 / (cutPoint * log(10))
+    offset = log10(cutPoint) - slope * cutPoint
+    gain = ei / nominalEI
+    gray = midGraySignal / gain
+    encGain = (log(gain) / log(2) * (0.89 - 1) / 3 + 1) * encodingGain
+    encOffset = encodingOffset
+
+    for i in range(0, 3):
+        nz = ((95 / 1023 - encOffset) / encGain - offset) / slope
+        encOffset = encodingOffset - log10(1 + nz) * encGain
+
+    # derived "ALEXA Log C Curve Usage in VFX 09-Mar-17" style parameters
+    # compatible with CLF cameraLogToLin parameters
+    cut = (cutPoint - nz) * gray * (0.18 / (midGraySignal * nominalEI / ei))
+    a = 1.0 / (gray * 0.18 / (midGraySignal * nominalEI / ei))
+    b = nz
+    c = encGain
+    d = encOffset
+
+    # print informative text if requested
+    if info:
+
+        # compute unused variables for completeness
+
+        # follows CLF 6.6 Log specification
+        e = c * (a / ((a * cut + b) * log(10)))
+
+        # follows CLF 6.6 Log specification
+        f = c * log10(a * cut + b) + d
+
+        print(f"White Paper Values for EI{ei:d}")
+        print(f"cut: {cut:.6f} :: {cut:.15f}")
+        print(f"a:   {a:.6f} :: {a:.15f}")
+        print(f"b:   {b:.6f} :: {b:.15f}")
+        print(f"c:   {c:.6f} :: {c:.15f}")
+        print(f"d:   {d:.6f} :: {d:.15f}")
+        print(f"e:   {e:.6f} :: {e:.15f}")
+        print(f"f:   {f:.6f} :: {f:.15f}")
+
+    # OCIO LogCameraTransform translation variables
+    base = 10.0
+    linSideBreak = [cut] * 3
+    logSideSlope = [c] * 3
+    logSideOffset = [d] * 3
+    linSideSlope = [a] * 3
+    linSideOffset = [b] * 3
+    direction = ocio.TRANSFORM_DIR_INVERSE
+
+    return transform_factory(
+        transform_type="LogCameraTransform",
+        transform_factory="Constructor",
+        base=base,
+        linSideBreak=linSideBreak,
+        logSideSlope=logSideSlope,
+        logSideOffset=logSideOffset,
+        linSideSlope=linSideSlope,
+        linSideOffset=linSideOffset,
+        direction=direction,
+    )
+
+
+def _build_awg4_mtx():
+    """
+    Build the `MatrixTransform` for ARRI Wide Gamut 4 primaries.
+
+    Values are copied directly from the LogC4 Specification.
+
+    Returns
+    -------
+    ocio.MatrixTransform
+         *OpenColorIO* `MatrixTransform`.
+    """
+
+    # Values as defined in: "ARRI LogC4 Logarithmic Color Space Specification".
```

Comment:
These values have more significant digits than what is published in the May 1, 2022 LogC4 specification.

Ideally, you could use matrix_RGB_to_RGB_transform, as you do above.  But it looks like AWG4 is not in colour yet, although perhaps Thomas could make a quick fix to unblock that?

The values you have here are ok, but I think they are very slightly different than what colour will give, once the AWG4 chromaticities are added.

Reply:
The intention is to match ARRI processing and have consistency across all LogC4 implementations. The white paper will be updated with these values, and it matches the official ACES IDT.

Reply:
Sounds good! 

The differences are really in the last digit or two of the double precision values, and the composition of your matrix and mine are within 2.7e-16 of an identity.  So It's ok for OCIO since the difference is well below the tolerance for optimization.